src/rgw/driver/d4n/rgw_sal_d4n.cc

```diff
@@ -832,6 +832,27 @@ int D4NFilterObject::D4NFilterReadOp::prepare(optional_yield y, const DoutPrefix
       ldpp_dout(dpp, 0) << "D4NFilterObject::" << __func__ << "(): failed to cache head object during eviction, ret=" << ret << dendl;
     }
   } else {
+    bufferlist etag_bl;
```

Comment:
this I think should only be done, after get_obj_attrs_from_cache(), otherwise the call to backend store should take care of it.

Reply:
Yes, the call to `get_obj_attrs_from_cache` is at the beginning of this if-else statement so once this `else` statement is reached, that method would have already been executed

Reply:
ok this piece of code looks correct